### PR TITLE
Fix clear button in Search field

### LINKED ISSUE #117: Search "clear" button does not work
If I want to clear a search phrase by pressing a "clear" button there is no additional request to fetch all the posts

### PR DESCRIPTION
**Pull request by michalklejdysz**
fix #117

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 .DS_Store
+
+# congfig
+/.idea
```
Comment (jkisiolek) [inline .gitignore:4]: Please set your IDE to add a newline at the end of the file

File: frontend/src/tests/utils/renderWithRouter.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import configureStore from 'redux-mock-store'
+import { createMemoryHistory } from 'history'
+import { render } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import { Route, Router, Switch } from 'react-router-dom'
+import React from 'react'
+
+const renderWithRouter = (component, historyPathArray = ['/']) => {
```
Comment (jkisiolek) [inline frontend/src/tests/utils/renderWithRouter.js:8]: Since its also provides a store, consider renaming to `renderWithStoreAndRouter`

**Review by jkisiolek — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (michalklejdysz) [reply .gitignore]: Solved

File: frontend/src/tests/utils/renderWithRouter.js (hunk 2 above)
Comment (michalklejdysz) [reply frontend/src/tests/utils/renderWithRouter.js]: :ok_hand: Renamed

**Review by jkisiolek — APPROVED** [review]

**Pull request merged**